components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -105,77 +134,7 @@ class GleanTest {
     }
 
     @Test
-    fun `test path generation`() {
-        val uuid = UUID.randomUUID()
-        val path = Glean.makePath("test", uuid)
-        val applicationId = "mozilla-components-service-glean"
-        // Make sure that the default applicationId matches the package name.
-        assertEquals(applicationId, Glean.applicationId)
-        assertEquals(path, "/submit/$applicationId/test/${Glean.SCHEMA_VERSION}/$uuid")
-    }
-
-    @Test
-    fun `test sending of default pings`() {
-        val server = MockWebServer()
-        server.enqueue(MockResponse().setBody("OK"))
-
-        StringsStorageEngine.clearAllStores()
-        ExperimentsStorageEngine.clearAllStores()
-        val stringMetric = StringMetricType(
-            disabled = false,
-            category = "telemetry",
-            lifetime = Lifetime.Application,
-            name = "string_metric",
-            sendInPings = listOf("default")
-        )
-
-        resetGlean(getContextWithMockedInfo(), Glean.configuration.copy(
-            serverEndpoint = "http://" + server.hostName + ":" + server.port,
-            logPings = true
-        ))
-
-        try {
-            stringMetric.set("foo")
-
-            Glean.setExperimentActive(
-                "experiment1", "branch_a"
-            )
-            Glean.setExperimentActive(
-                "experiment2", "branch_b",
-                    mapOf("key" to "value")
-            )
-            Glean.setExperimentInactive("experiment1")
-
-            Glean.handleEvent(Glean.PingEvent.Default)
-
-            val request = server.takeRequest()
-            assertEquals("POST", request.method)
-            val metricsJsonData = request.body.readUtf8()
-            val metricsJson = JSONObject(metricsJsonData)
-            checkPingSchema(metricsJson)
-            assertEquals(
-                "foo",
-                metricsJson.getJSONObject("metrics")
-                    .getJSONObject("string")
-                    .getString("telemetry.string_metric")
-            )
-            assertNull(metricsJson.opt("events"))
-            assertNotNull(metricsJson.opt("ping_info"))
-            assertNotNull(metricsJson.getJSONObject("ping_info").opt("experiments"))
-            val applicationId = "mozilla-components-service-glean"
-            assert(
-                request.path.startsWith("/submit/$applicationId/metrics/${Glean.SCHEMA_VERSION}/")
-            )
```

Comment:
I worry that the validation of the actual ping content is being lost here.  (Unless it was moved somewhere and I just can't find it).  I don't care too much about testing with a mock HTTP server and making sure pings get sent over HTTP (we can assume the HTTP client is working -- we didn't write it :wink:).  But it would be good to check that:

1) If an event is triggered, the expected pings are queued (since there might be more than one).
2) The contents of the pings contains what we expect them too
3) The contents of the pings validates against the schema

If an event is fired that generates and schedules some pings, is it true that those pings would be sitting on disk if there's nothing in the test that explicitly sends them out?  If so, maybe it would it be possible to add a testing-only function to the `PingStorageEngine` that returns pings that have been queued.  Something like:

`testGetQueuedPing(pingName: string): List<String>` (or `List<JSONObject>`)

This would go through all of the queued pings for the given ping type and return their contents.

Reply:
Some of this is broken out into other places now. You may have noticed the `isWorkScheduled()` helper function I use in the `PingSchedulerTest`.  What we did lose was the checking of the ping contents and schema.  I totally agree with the idea of sanity checking those and I'm working on a way to do that now.